### ISSUE TITLE
Guard 'conceal' in AgGroup/tests for vim < v7.3

### ISSUE THREAD

**Issue by amerlyq** [albfan/ag.vim#56]
Problem as description tells.

```
(2/7) [EXECUTE] Find "fuga" inside t/fixture directory
(2/7) [EXECUTE] (X) Vim(syntax):E475: Invalid argument: agLine /^\d\+:\d\+:/ conceal
```

**Comment by albfan** [comment]
Have to check and fix.

About vim 7.3 there's a matrix feature on travis and addons can specify version.

There are chances that:

```
matrix:
...
  haxe:
    addon: vim "7.3"
```

works out of the box.

**Comment by amerlyq** [comment]
Hm, I can't see your point. There truly are addons and they allow to have different set of installed software in matrix build. But it's impossible to have several vim builds in ubuntu precise. Moreover, all of them must be contained in whitelist of travis. Twice as impossible. Therefore I see no option beside building vim on CI by yourself, if you need to test on different versions. Because you can't simply add ppa for dockers on ubuntu. It's not Arch after all. 

There even no neovim package to test our plugin when async jobs will be added.
So I have to do all of this manually: in neovim I work and launch tests, when old vim will be tested locally in lucid chroot once on big merge. Have ideas to propose anything better?

**Comment by albfan** [comment]
See this

https://docs.travis-ci.com/user/firefox/

explains how to install different versions of firefox, Is it possible same flexibility for vim?

Reality is that if we want to test different vim versions or even different vim flavors (like neovim) we need to compile (4 minutes max as I say) or install different vim versions.

I will try some configs this weekend and create some branchs with different options. It is not neccesary to adopt one of that solutions, but at least discuss working configs.

**Comment by amerlyq** [comment]
Alright. I'm waiting on. Hope this will work as is.

**Comment by albfan** [comment]
I cannot make run travis on matrix. 

https://github.com/albfan/ag.vim/tree/matrix-travis

I will send an email to travis support team

but using several branches I can test neovim and vim73

https://github.com/albfan/ag.vim/tree/vim73
https://github.com/albfan/ag.vim/tree/neovim

An I cannot reproduce 

> (2/7) [EXECUTE] Find "fuga" inside t/fixture directory
> (2/7) [EXECUTE](X) Vim(syntax):E475: Invalid argument: agLine /^\d+:\d+:/ conceal

https://travis-ci.org/albfan/ag.vim/builds/97826351

can you show detailed version for vim?

**Comment by albfan** [comment]
Travis team have tell me how to generate different jobs on different travis environments from a single branch, so we can test all this stuff from our matrix of vim versions and ag versions. I will upload all, this afternoon.

**Comment by albfan** [comment]
Matrix build is working, I'm asking travis team to use yaml references

http://stackoverflow.com/questions/4150782/using-yaml-with-variables

to minimize travis config as nvim and vim73 environments need to be tested against same ag version as others.

**Comment by amerlyq** [comment]
Vim with that problem is standart for ubuntu 12.04 lucid:

``` yaml
VIM - Vi IMproved 7.2 (2008 Aug 9, compiled Nov 15 2012 00:31:26)
Included patches: 1-330
Compiled by buildd@
Huge version with GTK2 GUI.  Features included (+) or not (-):
+arabic +autocmd +balloon_eval +browse ++builtin_terms +byte_offset +cindent
+clientserver +clipboard +cmdline_compl +cmdline_hist +cmdline_info +comments
+cryptv +cscope +cursorshape +dialog_con_gui +diff +digraphs +dnd -ebcdic
+emacs_tags +eval +ex_extra +extra_search +farsi +file_in_path +find_in_path
+float +folding -footer +fork() +gettext -hangul_input +iconv +insert_expand
+jumplist +keymap +langmap +libcall +linebreak +lispindent +listcmds +localmap
+menu +mksession +modify_fname +mouse +mouseshape +mouse_dec +mouse_gpm
-mouse_jsbterm +mouse_netterm -mouse_sysmouse +mouse_xterm +multi_byte
+multi_lang -mzscheme +netbeans_intg -osfiletype +path_extra +perl +postscript
+printer +profile +python +quickfix +reltime +rightleft +ruby +scrollbind
+signs +smartindent -sniff +startuptime +statusline -sun_workshop +syntax
+tag_binary +tag_old_static -tag_any_white +tcl +terminfo +termresponse
+textobjects +title +toolbar +user_commands +vertsplit +virtualedit +visual
+visualextra +viminfo +vreplace +wildignore +wildmenu +windows +writebackup
+X11 -xfontset +xim +xsmp_interact +xterm_clipboard -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -D_REENTRANT -I/usr/include/gtk-2.0 -I/usr/lib/gtk-2.0/include -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/directfb -I/usr/include/libpng12     -Wall -g -O2    -D_REENTRANT -D_GNU_SOURCE -DDEBIAN  -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64  -I/usr/lib/perl/5.10/CORE  -I/usr/include/python2.6 -pthread -I/usr/include/tcl8.4  -D_REENTRANT=1  -D_THREAD_SAFE=1  -D_LARGEFILE64_SOURCE=1  -I/usr/lib/ruby/1.8/i486-linux
Linking: gcc   -L. -Wl,-Bsymbolic-functions -rdynamic -Wl,-export-dynamic  -Wl,-E  -Wl,--as-needed -o vim   -pthread -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lgdk_pixbuf-2.0 -lpangocairo-1.0 -lcairo -lpango-1.0 -lfreetype -lfontconfig -lgobject-2.0 -lgmodule-2.0 -lgthread-2.0 -lrt -lglib-2.0   -lXt -lncurses -lselinux  -lacl -lgpm -Wl,-E  -fstack-protector -L/usr/local/lib  -L/usr/lib/perl/5.10/CORE -lperl -L/usr/lib/python2.6/config -lpython2.6 -lutil -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions -L/usr/lib -ltcl8.4 -lieee -lruby1.8 -lrt -lm
```

**Comment by albfan** [comment]
I will try that exactly version but on travis vim 7.3 is working. (Running on a ubuntu 12.04)

Have to say that updating packages it install vim 7.4 so is yours an outdated vim package?

See apt-get command order on travis for vim 7.3

**Comment by albfan** [comment]
On ubuntu 12.04 it is vim 7.3 1-429

https://travis-ci.org/albfan/ag.vim/jobs/98645443

Are you enable to upgrade to that vim version?

As you can see I have merged matrix config to .travis. The only thing is left to do is decide what ag versions should be supported on neovim and vim73

**Comment by albfan** [comment]
http://packages.ubuntu.com/precise/vim-gtk seems to be 7.3 1-429, if you have more information please reopen this issue

**Comment by amerlyq** [comment]
Oops, I have checked and you were right. One of choorts is Ubuntu 10.04 and problem was exactly from it. Alright, anyways it's now my task to fix rewritten syntax file.

**Comment by amerlyq** [comment]
About neovim -- seems like we can't make it silent execute?

**Comment by albfan** [comment]
Really? Is that because neovim is under development or it will never be possible?

**Comment by amerlyq** [comment]
Temporary there is no way for neovim. Those features are broken, according to one of issue thread. But after v0.2 maybe they will be back as silent execution as well. But not now, a pity.
